src/Elan.Account/Services/UserProfileService.cs

```diff
@@ -31,6 +31,15 @@ public async Task<ElanUser> UpdateProfile(UserProfileViewModel model)
             user.FirstName = model.FirstName;
             user.LastName = model.LastName;
 
+            if (user.Gender != 0 && user.Gender != model.Gender)
+            {
+                throw new InvalidDataException("Gender cannot be changed.");
+            }
+            if (user.Gender == 0)
+            {
+                user.Gender = model.Gender;
+            }
+            
```

Comment:
something very important is missing here...

Reply:
fixed